docs/ldml/tr35.md

```diff
@@ -2972,6 +3033,40 @@ The binary operators '&', '-', and the implicit union have equal precedence and
 
 **One caution:** the '&' and '-' operators operate between sets. That is, they must be immediately preceded and immediately followed by a set. For example, the pattern **[[:Lu:]-A]** is illegal, since it is interpreted as the set **[:Lu:]** followed by the incomplete range **-A**. To specify the set of upper case letters except for 'A', enclose the 'A' in brackets: **[[:Lu:]-[A]]**.
 
+##### <a name="Variables_in_UnicodeSets" href="#Variables_in_UnicodeSets">Variables in UnicodeSets</a>
+
+Support for variable identifiers (var) is optional. 
+They are used in certain contexts such as in [https://cldr-smoke.unicode.org/spec/main/ldml/tr35-general.html#Transforms](Transforms).
+
+UnicodeSets may contain variables (`$my_char`, `$the_set`, ...) in place of full UnicodeSets and strings/characters.
+
+This introduces ambiguities when parsing. For example, it is unclear whether **[$a-$b]** is a range of strings/characters, or a set difference, because that depends on the values of the variables.
```

Comment:
We have to define it so that it **doesn't produce ambiguities**. Then this first line is false. That is, implementations that support variables must supply the set of variables and their values at the time that a UnicodeSet is parsed. (you could then explain why this is needed in a note, if you want).

Reply:
Thank you, agreed. I switched the order around such that I first require variables' values be known, and then I state the problems of the alternative in a conditional phrase.

Reply:
The definition is fine, and indeed unambiguous when everything is taken together; what this is trying to get at is that if you are solely doing parsing based on the grammar given above, you get a conflict which you can only resolve by looking at what the variables actually are.
This is not uncommon (e.g. `a * b;` in C could be declaring a variable `b` of tye `a*` or multiplying `a` by `b` and discarding that, depending on whether `a` is a type), but noteworthy.

```suggestion
The types of the variables need to be known in order to determine how to parse some UnicodeSets.
For example, the UnicodeSet **[$a-$b]** could formally be either a range if $a and $b are characters, or a set difference if they are sets.
In practice, it is expected that parsing would be done with the values of the variables already known. 
```

Reply:
@eggrobin I think we wrote over each other - Is the new version okay with you or too strict? https://github.com/unicode-org/cldr/pull/3089/files#diff-549880ab02f2fc27e87768d55cd7030b3afb0f5b18db21ef4297c2cddf9c985bR3047

Reply:
The types of the variables need to be known in order to determine how to parse some UnicodeSets.

Not just "in practice". And that needs to be the first point (before 'types of')

Reply:
(That suggested text is outdated.)